shared/src/main/kotlin/com/egm/stellio/shared/model/QueryParams.kt

```diff
@@ -3,9 +3,14 @@ package com.egm.stellio.shared.model
 import java.net.URI
 
 data class QueryParams(
-    val id: List<URI>? = null,
-    val expandedType: String? = null,
+    val ids: Set<URI>? = null,
+    val types: Set<ExpandedTerm>? = null,
     val idPattern: String? = null,
     val q: String? = null,
-    val expandedAttrs: Set<String> = emptySet()
+    val limit: Int,
+    val offset: Int,
+    val count: Boolean = false,
+    val attrs: Set<ExpandedTerm> = emptySet(),
+    val includeSysAttrs: Boolean = false,
+    val useSimplifiedRepresentation: Boolean = false
```

Comment:
as said above, as it is parsed, `types` will never be null, but empty if no `type` parameter is in the request. this is good news as it can be aligned with `attrs` which defaults to `emptySet()`.

and, for full consistency, `ids` should adopt this default :)